### PR TITLE
feat(tari-dan-cli): initial version

### PR DESCRIPTION
**Pull request by ksrichard**
Description
---
This is the initial version of Tari DAN CLI that includes the following:
- Automatic CLI configuration creation/restoration at every command run
- Configurable templates repository
- Automatically keep up-to-date locally the templates repository (if changed, just checks out and uses that, so multiple template repositories can be in place that can be switched by changing config)
- Generating new project (basically empty Cargo workspace project with some extra detail)
- Generating new WASM project (if generating into a Cargo project, it updates it's Cargo.toml automatically to include new project as workspace member)

Motivation and Context
---
The developers creating new templates needs a simple but powerful tool (a CLI in our case) to easily generate new projects, wasm projects, test them and deploy them with simple commands.

How Has This Been Tested?
---
Going through all CLI commands, create project, create multiple wasm projects under that etc..
NOTE: until https://github.com/tari-project/wasm-template/pull/9 gets merged, you should (after running any command there will be the config for CLI, on MAC: `/Users/<username>/Library/Application Support/tari_dan_cli/tari.config.toml`) update config to use a specific repo and branch:
```toml
[project_template_repository]
url = "https://github.com/ksrichard/wasm-template"
branch = "feature/tari-dan-cli-updates"
folder = "project_templates"

[wasm_template_repository]
url = "https://github.com/ksrichard/wasm-template"
branch = "feature/tari-dan-cli-updates"
folder = "wasm_templates"
```

What process can a PR reviewer use to test or verify this change?
---

Breaking Changes
---

- [x] None
- [ ] Requires data directory to be deleted
- [ ] Other - Please specify

### CODE REVIEW

**Comment by sdbondi** [comment]
LGTM - looks like something unrelated to this PR has gone wrong with CI and hickory client used in L1. Will see if I can fix it first thing tomorrow, if all is good will review and merge.

File: Cargo.toml [hunk 1]
```diff
@@ -56,7 +57,7 @@ members = [
     "utilities/transaction_submitter",
     "utilities/transaction_submitter",
     "utilities/generate_ristretto_value_lookup",
-    "applications/tari_watcher",
+    "applications/tari_watcher", "applications/tari_dan_cli",
```
Comment (sdbondi) [inline Cargo.toml]: Duplicate

**Review by sdbondi — COMMENTED** [review]
I think we should call this `tari_cli`  (binary will be `tari`) DAN is not a term we want to use in future

EDIT: I see you already named the binary `tari` :ok_hand: the crate can be `tari_cli`

**Comment by ksrichard** [comment]
> I think we should call this `tari_cli` (binary will be `tari`) DAN is not a term we want to use in future

okay, makes sense, let me update that

**Comment by ksrichard** [comment]
Closing, because the development has been moved to https://github.com/ksrichard/tari-cli

**Pull request closed without merging**